mypy/main.py

```diff
@@ -416,6 +424,100 @@ def get_init_file(dir: str) -> Optional[str]:
     return None
 
 
+# For most options, the type of the default value set in options.py is
+# sufficient, and we don't have to do anything here.  This table
+# exists to specify types for values initialized to None or container
+# types.
+config_types = {
+    # TODO: Check validity of python version
+    'python_version': lambda s: tuple(map(int, s.split('.'))),
+    'strict_optional_whitelist': lambda s: s.split(),
+    'custom_typing_module': str,
+}
+
+
+def parse_config_file(options: Options, filename: str) -> None:
+    """Parse a config file into an Options object.
+
+    Errors are written to stderr but are not fatal.
+    """
+    parser = configparser.RawConfigParser()
+    try:
+        parser.read(filename)
+    except configparser.Error as err:
+        print("%s: %s" % (filename, err), file=sys.stderr)
+        return
+    if 'mypy' not in parser:
+        print("%s: No [mypy] section in config file" % filename, file=sys.stderr)
+        return
+
+    section = parser['mypy']
+    prefix = '%s: [%s]' % (filename, 'mypy')
+    updates, report_dirs = parse_section(prefix, options, section)
+    for k, v in updates.items():
+        setattr(options, k, v)
+    options.report_dirs.update(report_dirs)
+
+    for name, section in parser.items():
+        if name.startswith('mypy-'):
+            glob = name[5:]
+            prefix = '%s: [%s]' % (filename, name)
+            updates, report_dirs = parse_section(prefix, options, section)
+            # TODO: Limit updates to flags that can be per-file.
+            if report_dirs:
+                print("%s: Per-file sections should not specify reports (%s)" %
+                      (prefix, ', '.join(s + '_report' for s in sorted(report_dirs))),
+                      file=sys.stderr)
+            if set(updates) - Options.PER_FILE_OPTIONS:
+                print("%s: Per-file sections should only specify per-file flags (%s)" %
+                      (prefix, ', '.join(sorted(set(updates) - Options.PER_FILE_OPTIONS))),
+                      file=sys.stderr)
+            options.per_file_options[glob] = updates
+
+
+def parse_section(prefix: str, template: Options,
```

Comment:
It could be useful to be able to specify `MYPYPATH` in the config file, maybe as `path = dir`.

What about excludes? If we had exclude globs, it would be possible to do simple blacklisting of some files/directories and then just run `mypy .`.

(No need to implement the above suggestions in this PR even if you think that they would be useful.)